lib/card/set_pattern.rb

```diff
@@ -130,6 +135,34 @@ def rule_set_key
         [ @anchor_id, self.class.pattern_code ].map( &:to_s ) * '+'
       end
     end
-  
+  end
+
+  class TypeSet < SetPattern
+    def initialize card
+      super
+      @inherit_card = card unless module_key
+    end
+
+    def lookup_inherited_key
+      return unless @inherit_card
+      card, @inherit_card = @inherit_card, nil
+
+      default_rule = card.rule_card(:default) and
+        type_code = default_rule.type_code and
+        #default_rule.cardname.size > 2 and
+        #default_rule.left.right.codename == self.class.pattern_code
+        mod_key = "Type::#{type_code.to_s.camelize}" and
+        ( Card::Set.modules[:nonbase_format].values +
+          [Card::Set.modules[:nonbase]] ).any?{|hash| hash[mod_key]} and
+        mod_key
+    end
+
+    def inherited_key
+      (defined? @inherited_key) ? @inherited_key : @inherited_key = lookup_inherited_key
+    end
+
+    def lookup_module_list modules_hash
```

Comment:
This is now an override in the Type subclass, so the inherited_key stuff is only use here and only defined here.